backend/controller/role.go

```diff
@@ -0,0 +1,98 @@
+package controller
+
+import (
+	"api/db"
+	"api/helpers"
+	"api/models"
+	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+)
+
+func GetAllRoles(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	err, roles := helpers.GetAllTable(w, []models.Role{})
+	if err == nil {
+		helpers.JsonWriter(w, roles)
+	}
+
+	//for viewing all connections
+	//roles = []models.Role{}
+	//config.DB.Preload("Doors").Find(&roles)
+	//helpers.JsonWriter(w, roles)
+}
+
+func CreateRole(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var role models.Role
+	err := json.NewDecoder(r.Body).Decode(&role)
+	role.Common = models.Common{}
```

Comment:
I'm still confused by this, maybe you can explain better here: does instantiating a model not populate this by default?

What values get populated into common? Does it default CreatedAt and UpdatedAt to the current time? Isn't the Id set when the model is created (this is the main reason I'm asking about this), so what functionality does this add?

Also maybe we can write a `BeforeCreate` for our models to specify it there instead of here? If this sounds like a good approach, let's open up a follow up PR for this.

Reply:
A before create maybe, but essentially I just dont want users to be able to called the funciton with request body as follows
{
"id": 10000,
"CreatedAt": (some time maybe yesterday),
"Name": John,
...
}

Essentially this just resets those common values to default to ignore any population a user would attempt to do since gorm should be the only thing populating those fields and since we cant set those fields to read only. Also since looping over fields in goLang is like annoying cuz of reflection this avoids having to do any looping. But perhaps this could be done in a before create, could be a clean way didnt think of that